src/Executor/Runner/Docker.php

```diff
@@ -0,0 +1,1131 @@
+        });
+
+        Console::success('Maintenance interval started.');
+
+        /**
+         * Get usage stats every X seconds to update swoole table
+         */
+        Console::info('Starting stats interval...');
+        $getStats = function (Table $statsHost, Table $statsContainers): void {
+            // Get usage stats
+            $usage = new Usage($this->orchestration);
+            $usage->run();
+
+            // Update host usage stats
+            if ($usage->getHostUsage() !== null) {
+                $oldStat = $statsHost->get('host', 'usage') ?? null;
+
+                if ($oldStat === null) {
+                    $stat = $usage->getHostUsage();
+                } else {
+                    $stat = ($oldStat + $usage->getHostUsage()) / 2;
+                }
+
+                $statsHost->set('host', ['usage' => $stat]);
+            }
+
+            // Update runtime usage stats
+            foreach ($usage->getRuntimesUsage() as $runtime => $usageStat) {
+                $oldStat = $statsContainers->get($runtime, 'usage') ?? null;
+
+                if ($oldStat === null) {
+                    $stat = $usageStat;
+                } else {
+                    $stat = ($oldStat + $usageStat) / 2;
+                }
+
+                $statsContainers->set($runtime, ['usage' => $stat]);
+            }
+
+            // Delete gone runtimes
+            $runtimes = \array_keys($usage->getRuntimesUsage());
+            foreach ($statsContainers as $hostname => $stat) {
+                if (!(\in_array($hostname, $runtimes))) {
+                    $statsContainers->delete($hostname);
+                }
+            }
+        };
+
+        // Load initial stats in blocking way
+        $getStats($statsHost, $statsContainers);
+
+        // Setup infinite recurssion in non-blocking way
+        \go(function () use ($getStats, $statsHost, $statsContainers) {
+            Timer::after(1000, fn () => $getStats($statsHost, $statsContainers));
+        });
+
+        Console::success('Stats interval started.');
+
+        Process::signal(SIGINT, fn () => $this->cleanUp($activeRuntimes, $networks));
+        Process::signal(SIGQUIT, fn () => $this->cleanUp($activeRuntimes, $networks));
+        Process::signal(SIGKILL, fn () => $this->cleanUp($activeRuntimes, $networks));
+        Process::signal(SIGTERM, fn () => $this->cleanUp($activeRuntimes, $networks));
+    }
+
+    /**
+     * @param string $runtimeId
+     * @param int $timeout
+     * @param Response $response
+     * @param Log $log
+     * @return void
+     */
+    public function getLogs(string $runtimeId, int $timeout, Response $response, Log $log): void
+    {
+        $runtimeName = System::getHostname() . '-' . $runtimeId;
+
+        // Wait for runtime
+        for ($i = 0; $i < 10; $i++) {
+            $output = '';
+            $code = Console::execute('docker container inspect ' . \escapeshellarg($runtimeName), '', $output);
```

Comment:
strange I thought we used the Docker API here? Doesn't the `Docker()` adapter have this method?